kickstarter/android-oss
# NT-1955: UI – Comment Card with threads

# 📲 What

UI – Comment Card with threads

# 🤔 Why

- In order to support threads we need to add the missing controls to our Comment
- Also, we need to support yet another variation of the Comment Card design, to display replies

# 🛠 How

- Added padding end and margin guidelines to threadactivitylayout
```
<androidx.constraintlayout.widget.Guideline
            android:id="@+id/left_guideline"
            app:layout_constraintGuide_begin="@dimen/grid_5"
            android:orientation="vertical"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"/>


        <androidx.recyclerview.widget.RecyclerView
            android:id="@+id/comment_replies_recycler_view"
            app:layout_constraintStart_toEndOf="@id/left_guideline"
            android:layout_width="0dp"
            android:paddingEnd="@dimen/grid_3"
            android:layout_height="wrap_content"
            android:background="@color/kds_white"
            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
            app:layout_constraintTop_toBottomOf="@id/comments_card_view"
            tools:layout_editor_absoluteX="0dp"
            tools:listitem="@layout/comment_card" />
```

# 👀 See

![Screen Shot 2021-05-30 at 12 37 55 PM](https://user-images.githubusercontent.com/63934292/120105903-b8646b80-c152-11eb-87a6-e9f7237534fc.png)

<img width="263" alt="Screen Shot 2021-05-30 at 2 19 06 PM" src="https://user-images.githubusercontent.com/63934292/120105908-bb5f5c00-c152-11eb-8ec8-24f55fbdad9b.png">


| Before 🐛 | After 🦋 |
| --- | --- |
|  |  |

# 📋 QA

- Select any project. Go to comments.  You should see the updated comment card
- Click ``` View replies ``` button. You should be taken to the threads screen

# Story 📖

https://kickstarter.atlassian.net/browse/NT-1955
